app/src/main/java/com/totsp/crossword/BackgroundDownloadService.java

```diff
@@ -0,0 +1,156 @@
+package com.totsp.crossword;
+
+import android.Manifest;
+import android.annotation.TargetApi;
+import android.app.NotificationManager;
+import android.app.job.JobInfo;
+import android.app.job.JobParameters;
+import android.app.job.JobScheduler;
+import android.app.job.JobService;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.os.AsyncTask;
+import android.os.Looper;
+import android.preference.PreferenceManager;
+import android.support.v4.content.ContextCompat;
+
+import com.totsp.crossword.net.Downloaders;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+// Currently only available on API version >=21 due to use of JobScheduler.
+// It may be possible to implement this functionality using AlarmManager for lower SDK versions.
+@TargetApi(21)
+public class BackgroundDownloadService extends JobService {
+    public static final String DOWNLOAD_PENDING_PREFERENCE = "backgroundDlPending";
+
+    private static final Logger LOGGER =
+            Logger.getLogger(BackgroundDownloadService.class.getCanonicalName());
+
+    private static JobInfo getJobInfo(boolean requireUnmetered, boolean allowRoaming,
+                                      boolean requireCharging) {
+        JobInfo.Builder builder = new JobInfo.Builder(
+                JobSchedulerId.BACKGROUND_DOWNLOAD.id(),
+                new ComponentName("com.totsp.crossword.shortyz",
+                        BackgroundDownloadService.class.getName()));
+
+        builder.setPeriodic(TimeUnit.HOURS.toMillis(1))
+                .setRequiredNetworkType(JobInfo.NETWORK_TYPE_UNMETERED)
```

Comment:
This would likely be better done at something like 07:00UTC. That generally means that the puzzles that update at Midnight pacific time will be there. Not a lot of reason to keep running all the DLs.